src/docker-extract.c

```diff
@@ -70,38 +89,54 @@ int apply_whiteout(const char *wh_marker, char *rootfs_dir) {
     }
 
     // Start with ROOTFS
-    retval = snprintf(buf, sizeof(buf), "%s/", rootfs_dir);
-    if (retval == -1 || retval >= sizeof(buf)) {
+    retval = snprintf(target, sizeof(target), "%s/", rootfs_dir);
+    if (retval == -1 || retval >= sizeof(target)) {
         singularity_message(ERROR, "Error with pathname too long\n");
         ABORT(255);
     }
-    l = strlen(buf);
+    l = strlen(target);
     // Add whiteout path up to .wh.
     token_pos = strlen(wh_marker) - strlen(token);
-    retval = snprintf(buf + l, token_pos + 1, "%s", wh_marker);
-    if (retval == -1 || retval >= sizeof(buf) - l) {
+    retval = snprintf(target + l, token_pos + 1, "%s", wh_marker);
```

Comment:
As mentioned in Slack, I think this could overflow with a sufficiently large `wh_marker` input. Add check to protect against that overflow.